### ISSUE TITLE
@nextcloud-bot is not able to push translations

### ISSUE THREAD

**Issue by korelstar** [stefan-niedermann/nextcloud-notes#196]
see https://github.com/stefan-niedermann/nextcloud-notes/pull/180#issuecomment-285417285:
> Currently the transifex push fails, because the nextcloud-bot get rejected with "protected branch hook declined".

**Comment by stefan-niedermann** [comment]
cc @MorrisJobke

**Comment by MorrisJobke** [comment]
> Currently the transifex push fails, because the nextcloud-bot get rejected with "protected branch hook declined".

Give him admin rights for this repo and I can do the rest.

**Comment by MorrisJobke** [comment]
@nextcloud-bot is the account

**Comment by stefan-niedermann** [comment]
Yeah, it broke build, therefore it was not able to merge into master...
How can i give admin rights to it? It is already "contributor"

**Comment by MorrisJobke** [comment]
Set it as admin:

![bildschirmfoto 2017-03-27 um 15 32 15](https://cloud.githubusercontent.com/assets/245432/24379057/9513112a-1302-11e7-8cbb-80d50d4d13c3.png)

**Comment by korelstar** [comment]
Travis says: `.../res/values-sq/strings.xml:84:44: Error: The entity name must immediately follow the '&' in the entity reference.`
Can someone fix this in Transifex? I have only access to German there.

**Comment by stefan-niedermann** [comment]
@MorrisJobke i think one can set this only in organizations.

![bildschirmfoto vom 2017-03-28 11-06-07](https://cloud.githubusercontent.com/assets/4741199/24397467/9e58d6b2-13a6-11e7-8040-fe1f5bc58638.png)

**Comment by MorrisJobke** [comment]
> @MorrisJobke i think one can set this only in organizations.

Seems so :/

**Comment by korelstar** [comment]
So none of you both (@MorrisJobke and @stefan-niedermann) changed something to fix this? Nevertheless, the bot recently was able to commit. Hence, the original problem is solved, for now (really without any intervention?).

Did you were able to see the error message I've posted above? I think this was the source of the problem: Travis blocked the commit because there is (again) a translation with syntax errors.

**Comment by MorrisJobke** [comment]
> Did you were able to see the error message I've posted above? I think this was the source of the problem: Travis blocked the commit because there is (again) a translation with syntax errors.

Is this stuff like this: https://github.com/stefan-niedermann/nextcloud-notes/commit/32782abafd2de3b2799c52e8bc7881ad0ee00e8b#diff-73f9c9346c3c9b3ffd5a81532f1b1848R85 ?

**Comment by korelstar** [comment]
The error was introduced in the first commit after the long outage: https://github.com/stefan-niedermann/nextcloud-notes/commit/b4590ba6f66bf362537ccd7792963617e0c9ee0d
It's in this line: https://github.com/stefan-niedermann/nextcloud-notes/blob/b4590ba6f66bf362537ccd7792963617e0c9ee0d/app/src/main/res/values-sq/strings.xml#L84

Travis wrote the following to his log, but I can see only the log for the "successful" commit (not for the failed attempts):

> .../res/values-sq/strings.xml:84:44: Error: The entity name must immediately follow the '&' in the entity reference.

**Comment by MorrisJobke** [comment]
@korelstar I fixed it on transifex and triggered a new sync of transifex. Could you check if all is fine now?

**Comment by stefan-niedermann** [comment]
> So none of you both (@MorrisJobke and @stefan-niedermann) changed something to fix this? 

It was me. i had a rule "only allow merge when travis build is successful" activated. then transifex stopped working. after you opemed this issue, i disabled the rule what allowed the bot to merge into master with a broken build.

**Comment by MorrisJobke** [comment]
> It was me. i had a rule "only allow merge when travis build is successful" activated. then transifex stopped working. after you opemed this issue, i disabled the rule what allowed the bot to merge into master with a broken build.

I can also push to a different branch if you want and you then just merge this branch from time to time ;) Then you could protect master again.

**Comment by stefan-niedermann** [comment]
It's maybe the best solution. We only need up2date translations before we release a new version.
The target is to get a always buildable `master` - with transifex i never know if something comes that breaks the build.
contra is that we have to remember each release to merge the translation updates back to `master`.

@korelstar opinion?

**Comment by korelstar** [comment]
I vote for committing to a separate branch and merging manually on demand. The protection of the master branch (build must be successful) is a nice feature that should be activated, IMHO.

**Comment by korelstar** [comment]
@MorrisJobke Thanks for fixing the syntax error. The build seems to be fine (according to Travis; I can't test it by myself today), but the HTML in that translation is still erroneous (weird spaces in tag). If your lucky, please fix this, too. But this error affects only a single language, so the respective translator can fix that translation eventually. I.e., this has a low priority.

**Comment by stefan-niedermann** [comment]
@MorrisJobke I created the branch `transifex` - could you please point to this branch as merge target?

**Comment by MorrisJobke** [comment]
> @MorrisJobke I created the branch transifex - could you please point to this branch as merge target?

I would create a new transifex branch every night and forcefully overwrite it. Is this okay?

**Comment by korelstar** [comment]
For me, that's fine - if it works technically.

**Comment by stefan-niedermann** [comment]
I'm fine with that, too :)

**Comment by stefan-niedermann** [comment]
Err. @MorrisJobke : With the current solution the branch `transifex` is not needed, right? So i can delete it and close this ticket?

**Comment by korelstar** [comment]
We will need the `transifex` branch for switching to the new approach discussed above. I already made a proposal for changing the respective script (see https://github.com/stefan-niedermann/nextcloud-notes/pull/180#issuecomment-285316069). Should I make a PR for a new shell script in the repository `nextcloud/docker-ci`, @MorrisJobke ?

**Comment by MorrisJobke** [comment]
> We will need the transifex branch for switching to the new approach discussed above. I already made a proposal for changing the respective script (see #180 (comment)). Should I make a PR for a new shell script in the repository nextcloud/docker-ci, @MorrisJobke ?

That would be nice. The branch is not really needed now, because it will be recreated once the bot pushes.

**Comment by MorrisJobke** [comment]
Could someone please un-protect the transifex branch. Otherwise I can't force push to this branch. ;) cc @korelstar @stefan-niedermann

**Comment by MorrisJobke** [comment]
Once this is done we can close this ticket here \o/

**Comment by stefan-niedermann** [comment]
done^^

**Comment by MorrisJobke** [comment]
Thanks - tested and works 👍

**Comment by stefan-niedermann** [comment]
thank you for help :)

**Comment by MorrisJobke** [comment]
> thank you for help :)

Thanks for your patience :) It took some time ... :/

**Comment by stefan-niedermann** [comment]
No problem. Using transifex at all is a great improvement compared to the 4-5 languages the app has been translated into until v0.8.0 ;-)

Anyway, there is no time pressure, we all work in our spare time. It's ready when it's ready.

**Comment by korelstar** [comment]
@MorrisJobke Thank you very much! It was just in the right moment: the latest changes on Transifex break the build again!
see https://travis-ci.org/stefan-niedermann/nextcloud-notes/builds/224172321

Hence, the decision to do this in a separate branch was definitely right. However, I'm wondering if and why these problems happens to this project more often than in other projects (e.g. nextcloud/android).

More important is the question how to proceed in these cases. We could ask you or somebody for help, but maybe it would be better if we have write access to all languages on Transifex?

**Comment by MorrisJobke** [comment]
> More important is the question how to proceed in these cases. We could ask you or somebody for help, but maybe it would be better if we have write access to all languages on Transifex?

Maybe ask @AndyScherzinger how he does it. ;)

**Comment by AndyScherzinger** [comment]
That is actually how I watch over the Nextcloud Android repo... Identify the language string that breaks the build, fix it on Transifex and trigger the bot. I can fix the translation since I have manager rights on Transifex and ping Mario or  Joas or else on irc to trigger the bot. 

So it also does happen every now and then for the other repos...

**Comment by MorrisJobke** [comment]
@korelstar Maybe also try to ping me, @nickvergessen or @jospoortvliet to fix the translations on transifex.
